ci/ci/github.py

```diff
@@ -896,21 +917,25 @@ async def _start_deploy(self, db: Database, batch_client):
 (cd {shq(repo_dir)}; {self.checkout_script()})
 '''
             )
+
+            token = generate_token()
             with open(f'{repo_dir}/build.yaml', 'r', encoding='utf-8') as f:
-                config = BuildConfiguration(self, f.read(), requested_step_names=DEPLOY_STEPS, scope='deploy')
-                namespace = config.namespace()
+                config = BuildConfiguration(
+                    self, f.read(), DEFAULT_NAMESPACE, token, requested_step_names=DEPLOY_STEPS, scope='deploy'
+                )
                 services = config.deployed_services()
             with open(f'{repo_dir}/ci/test/resources/build.yaml', 'r', encoding='utf-8') as f:
-                test_services = BuildConfiguration(self, f.read(), scope='deploy').deployed_services()
+                test_services = BuildConfiguration(
+                    self, f.read(), DEFAULT_NAMESPACE, token, scope='deploy'
+                ).deployed_services()
 
             services.extend(test_services)
-            assert namespace is not None
-            await add_deployed_services(db, namespace, services, None)
+            await add_deployed_services(db, DEFAULT_NAMESPACE, services)
 
             log.info(f'creating deploy batch for {self.branch.short_str()}')
             deploy_batch = batch_client.create_batch(
                 attributes={
-                    'token': secrets.token_hex(16),
+                    'token': token,
```

Comment:
This is a nice change. What purpose did the token attribute serve before? It seems useless?

Reply:
Ya I guess so